googletest/include/gtest/internal/gtest-type-util.h

```diff
@@ -85,6 +101,20 @@ inline std::string GetTypeName(const std::type_info& type) {
   const std::string name_str(status == 0 ? readable_name : name);
   free(readable_name);
   return CanonicalizeForStdLibVersioning(name_str);
+#elif defined(_MSC_VER)
+  // Strip struct and class due to differences between
+  // MSVC and other compilers. std::pair<int,bool> is printed as
+  // "struct std::pair<int,bool>" when using MSVC vs "std::pair<int, bool>" with
+  // other compilers.
+  std::string s = name;
+  // Only strip the leading "struct " and "class ", so uses rfind == 0 to 
+  // ensure that
+  if (s.rfind("struct ", 0) == 0) {
+    s = s.substr(strlen("struct "));
```

Comment:
Super pedantic nit: Let's use `std::strlen`

Reply:
the existing code in gtest-type-util.h are using `strlen`, not `std::strlen`, so I'll keep use strlen,
if std::strlen are needed, I think it's should in another MR and change all usage of `strlen` to `std::strlen`